pkg/cli/bicep/envinject.go

```diff
@@ -35,12 +35,13 @@ func InjectApplicationParam(deploymentTemplate map[string]any, parameters map[st
 }
 
 func injectParam(deploymentTemplate map[string]any, parameters map[string]map[string]any, parameter string, value string) error {
-	if deploymentTemplate["parameters"] == nil {
-		return nil
+	formalParams, err := ExtractParameters(deploymentTemplate)
```

Comment:
Could be just me - what do we mean by "formal" params?

Reply:
You need a term to tell the difference between parameters that were declared, and the values associated with those parameters.

- declaration: `param foo string`
- value `--parameters foo=bar`

Usually the declaration is called formal parameters.